src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Create.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+
+using Microsoft.Win32.SafeHandles;
+
+using Internal.Cryptography;
+
+using ErrorCode = Interop.NCrypt.ErrorCode;
+using NCRYPT_UI_POLICY = Interop.NCrypt.NCRYPT_UI_POLICY;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    ///     Managed representation of an NCrypt key
+    /// </summary>
+    public sealed partial class CngKey : IDisposable
+    {
+        //
+        // Creation factory methods
+        //
+
+        public static CngKey Create(CngAlgorithm algorithm)
+        {
+            return Create(algorithm, keyName: null);
+        }
+
+        public static CngKey Create(CngAlgorithm algorithm, string keyName)
+        {
+            return Create(algorithm, keyName, creationParameters: null);
+        }
+
+        public static CngKey Create(CngAlgorithm algorithm, string keyName, CngKeyCreationParameters creationParameters)
+        {
+            if (algorithm == null)
+                throw new ArgumentNullException("algorithm");
+
+            if (creationParameters == null)
+                creationParameters = new CngKeyCreationParameters();
+
+            SafeNCryptProviderHandle providerHandle = creationParameters.Provider.OpenStorageProvider();
+            SafeNCryptKeyHandle keyHandle;
+            ErrorCode errorCode = Interop.NCrypt.NCryptCreatePersistedKey(providerHandle, out keyHandle, algorithm.Algorithm, keyName, 0, creationParameters.KeyCreationOptions);
+            if (errorCode != ErrorCode.ERROR_SUCCESS)
+                throw errorCode.ToCryptographicException();
+
+            InitializeKeyProperties(keyHandle, creationParameters);
+
+            errorCode = Interop.NCrypt.NCryptFinalizeKey(keyHandle, 0);
+            if (errorCode != ErrorCode.ERROR_SUCCESS)
+                throw errorCode.ToCryptographicException();
+
+            CngKey key = new CngKey(providerHandle, keyHandle);
+
+            // No name translates to an ephemeral key
+            if (keyName == null)
+            {
+                key.IsEphemeral = true;
+            }
+
+            return key;
+        }
+
+        /// <summary>
+        ///     Setup the key properties specified in the key creation parameters
+        /// </summary>
+        private static void InitializeKeyProperties(SafeNCryptHandle keyHandle, CngKeyCreationParameters creationParameters)
+        {
+            unsafe
```

Comment:
I could but I stick consistently to the rule that "unsafe" only goes into the signature if one of the parameter types is unsafe. Otherwise, it's an implementation detail and it rubs me the wrong way to put the keyword into the signature line.